File: backend/app/schemas/file.py
"""File schemas."""

from __future__ import annotations

from typing import Optional

from pydantic import BaseModel


class FileResponse(BaseModel):
    id: str
    subject_id: str
    filename: str
    file_type: str
    file_size: int
    summary: Optional[str] = None
    page_count: Optional[int] = None
    created_at: str
    has_extracted_text: bool = False
    extraction_error: Optional[str] = None

    class Config:
        from_attributes = True


class FileListResponse(BaseModel):
    files: list[FileResponse]
    total: int

File: frontend/src/schemas/file.ts
import { z } from 'zod';

export const SourceFileSchema = z.object({
  id: z.string().uuid(),
  subject_id: z.string().uuid(),
  filename: z.string(),
  file_type: z.string(),
  file_size: z.number(),
  summary: z.string().nullable(),
  has_extracted_text: z.boolean(),
  page_count: z.number().nullable(),
  created_at: z.string().datetime(),
});
